ddl/table.go

```diff
@@ -56,6 +56,11 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
+	if tbInfo.Partition != nil && tbInfo.Partition.Enable && tbInfo.TempTableType == model.TempTableGlobal {
```

Comment:
```suggestion
	if tbInfo.Partition != nil {
```
No need to check `Enable`. It's a historical problem.
Besides, it also doesn't support local temporary tables in the future.